dash/testing/wait.py

```diff
@@ -107,10 +107,9 @@ def __call__(self, driver):
         try:
             elem = self._get_element(driver)
             logger.debug("text to equal {%s} => expected %s", elem.text, self.text)
-            return (
-                str(elem.text) == self.text
-                or str(elem.get_attribute("value")) == self.text
-            )
+            if (value := elem.get_attribute("value")) is not None:
+                return str(elem.text) == self.text or str(value) == self.text
+            return str(elem.text) == self.text
```

Comment:
```suggestion
            value = elem.get_attribute("value")
            return str(elem.text) == self.text or (value is not None and str(value) == self.text)
```

IMO better to avoid the walrus operator here as it's clearer without. Same goes for the other function.